Robot.java_03_22_17

```diff
@@ -1,518 +1 @@
-		}
-*/			
-	
-
-		if (autonomousCommand != null)
-			autonomousCommand.cancel();
-		
-		left.disable();
-		right.disable();
-		driveTrain = new RobotDrive(leftSpark, rightSpark);
-	}
-
-	/**
-	 * This function is called periodically during operator control
-	 */
-	
-	@Override
-	public void teleopPeriodic() {
-		
-		//needs a String/boolean parameter		
-		Scheduler.getInstance().run();
-		
-		//double v = s1.getDistance();
-
-		//	System.out.println(v);
-
-		System.out.println("LEFT\tgetDistance: \t\t" + leftEncoder.getDistance() + "\t\tleftRate: \t\t" + leftEncoder.getRate());
-		System.out.println("LEFT JOYSTICK" + -1*stick2.getY());
-		System.out.println("RIGHT\tgetDistance: \t\t" + rightEncoder.getDistance() + "\t\trightRate: \t\t" + rightEncoder.getRate());
-		System.out.println("RIGHT JOYSTICK" + -1*stick.getY());
-		System.out.println();
-		  if (stick2.getTrigger() && !prevTrigger) {
-			    System.out.println("Setting camera 1\n");
-			    server.setSource(cam1);
-			  }
-		  else if (!stick2.getTrigger() && prevTrigger) {
-			    System.out.println("Setting camera 0\n");
-			    server.setSource(cam0);
-			  } 
-		  prevTrigger = stick2.getTrigger();
-
-
-		
-		if (driveMode.equalsIgnoreCase("TANK")) {
-			driveTrain.tankDrive(-1*stick2.getY(), stick.getY());
-			//left, right
-		}
-		else if(driveMode.equalsIgnoreCase("ARCADE")){
-			driveTrain.arcadeDrive(-stick.getY(), -stick.getX());
-			//driveTrain.arcadeDrive(stick);
-		}
-		else if(driveMode.equalsIgnoreCase("BACK TANK")){
-			driveTrain.tankDrive(1*stick.getY(), -1*stick2.getY());
-		}
-		
-		//winch control - Halie 
-		winch.tankDrive(stick.getRawAxis(3), 0);
-		
-	}
-			
-
-	/**
-	 * This function is called periodically during test mode
-	 */
-	@Override
-	public void testPeriodic() {
-		LiveWindow.run();
-	}
-	 
-	public double feetToEncUnits(double feet) {
-		return feet*12/(6*Math.PI)*360/2400;
-	}
-	
-	
-	public void moveForward(double feet) {	
-		if (leftEncoder.getDistance() < feetToEncUnits(feet) || rightEncoder.getDistance() < feetToEncUnits(feet)) {
-			//rotations: X * 2400 / 360 
-			//feet: 
-			left.set(.5);
-			right.set(.5);
-		}
-		else {
-			left.set(0);
-			right.set(0);
-		} 
-	}
-	
-	public void moveForwardSlow(double feet) {
-		if (leftEncoder.getDistance() < feetToEncUnits(feet) || rightEncoder.getDistance() < feetToEncUnits(feet)) {
-			//rotations: X * 2400 / 360 
-			//feet: 
-			left.set(.1);
-			right.set(.1);
-		}
-		else {
-			left.set(0);
-			right.set(0);
-		} 
-	}
-	
-	
-	public boolean finishedMoving(double feet) {
-		if (leftEncoder.getDistance() < feetToEncUnits(feet) || rightEncoder.getDistance() < feetToEncUnits(feet)) {
-			return false;
-		}
-		else 
-			leftEncoder.reset();
-			rightEncoder.reset();
-			return true;
-	}
-	
-	public void changeAngle(double degrees) {
-		//want to turn until getAngle > degrees
-		//won't have to be exact 
-		
-		System.out.println(gyro.getAngle());
-		if (degrees >= 0) {
```

Comment:
I think the changeAngle will mostly work at this point. One potential issue I see is that you are writing the code at the top as if the turns are relative to the current location as opposed to an absolute heading (e.g. turn right 30 degrees as opposed to go to heading 70 degrees) but this code is written as if you're looking for a specific heading because you do not reset the gyro to zero when you start the turn. You can decide which approach makes the most sense and experiment during the practice day but:

If you want to do the absolute heading
- It might actually help because you can adjust for any drift that might happen in an earlier move forward step. So if you drifted 5 degrees, you would now only turn 25 degrees more to get a 30 degree turn.
- It will be a little trickier to think through the headings you want since it will all be relative to the robot's starting orientation.
- To take this approach, you need to change the code slightly when deciding to go left or right to take into account where you are now. If you're current heading is 75 degrees and you want to change to 30 degrees, you need to turn in the negative direction

If you want to do relative heading
- You could get some error from when you move forwards or make other turns. that error will accumulate
- You need to change the code to reset the gyro to zero when you start your turn

Things to keep in mind in both cases:
- You'll want to double check that positive is left on the gyro and negative is right
- I can't remember if the gyro will give you the heading MOD something, or if it keeps changing continuously. i.e., when you get to 361 degrees, does it read 361 or 1, or when it goes negative, does it go to 359?. You'll have to check and change your code appropriately to handle those cases so that you are always turning the shortest way to get what you want. This is more important if you choose the heading approach.